src/server/pfs/server/driver.go

```diff
@@ -438,35 +438,44 @@ func (d *driver) deleteRepos(ctx context.Context, projects []*pfs.Project) ([]*p
 }
 
 func (d *driver) deleteReposInTransaction(txnCtx *txncontext.TransactionContext, projects []*pfs.Project) ([]*pfs.Repo, error) {
-	var deleted []*pfs.Repo
-	var repoInfos []*pfs.RepoInfo
+	var ris []*pfs.RepoInfo
 	if err := d.listRepoInTransaction(txnCtx, false, "", projects, func(ri *pfs.RepoInfo) error {
-		ok, err := d.canDeleteRepo(txnCtx, ri.Repo)
-		if err != nil {
-			return err
-		} else if ok {
-			repoInfos = append(repoInfos, ri)
-		}
+		ris = append(ris, ri)
 		return nil
 	}); err != nil {
 		return nil, errors.Wrap(err, "list repos")
 	}
-	if len(repoInfos) == 0 {
+	return d.deleteReposHelper(txnCtx, ris, false)
+}
+
+func (d *driver) deleteReposHelper(txnCtx *txncontext.TransactionContext, ris []*pfs.RepoInfo, force bool) ([]*pfs.Repo, error) {
```

Comment:
Is this recursive?  Why is this a separate method if it's only called once and doesn't have a descriptive name?

Reply:
I see that this is also called from the other file.  It seems like `deleteReposInTransaction` and `DeleteReposInTransaction` are the same things?  When would we use one vs. the other?

Reply:
it's an internal method that multiple functions rely on. Does Helper imply recursive? Happy to call this deleteReposInternal()

Reply:
>  Does Helper imply recursive
It doesn't really imply anything.  It's not really descriptive, which is why I was trying to figure out why it was split out.

Either it has to be a separate function because it's recursive, or has multiple callers, or it's split out to make the code clearer.  With a name like `*Helper` it isn't really the latter, so it should be one of the first two.  I didn't see the multiple call sites at first.